src/main/java/seedu/address/model/application/PositionApplied.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.application;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Company's internship position in the application list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPosition(String)}
+ */
+public class PositionApplied {
+
+    public static final String MESSAGE_CONSTRAINTS = "Positions can take any values, and it should not be blank";
+
+    /*
+     * The first character of the position applied must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
```

Comment:
I think this regex should be the same as CompanyName since they are basically both names.